mypyc/genops.py

```diff
@@ -460,8 +484,18 @@ def prepare_class_def(path: str, module_name: str, cdef: ClassDef,
         elif isinstance(node.node, (FuncDef, Decorator)):
             prepare_method_def(ir, module_name, cdef, mapper, node.node)
         elif isinstance(node.node, OverloadedFuncDef):
-            assert node.node.impl
-            prepare_method_def(ir, module_name, cdef, mapper, node.node.impl)
+            # Handle case for property with both a getter and a setter
+            if node.node.is_property:
+                if is_valid_multipart_property_def(node.node):
+                    for item in node.node.items:
+                        prepare_method_def(ir, module_name, cdef, mapper, item)
+                else:
+                    errors.error("Unsupported property decorator semantics", path, cdef.line)
```

Comment:
Add a test for this error.

Reply:
I can't actually get this check to fail because the things that it checks are essentially the same thing that mypy checks for in semanal, so whenever I try to fail this check, I get:
"error: Decorated property not supported"
from mypy, which is the correct error message. I figure I should still keep the check in here if anything ever changes?